packages/workspace/src/tasks-runner/task-orchestrator.ts

```diff
@@ -272,9 +348,12 @@ export class TaskOrchestrator {
           }
           // we don't have to worry about this statement. code === 0 guarantees the file is there.
           if (outputPath && code === 0) {
+            const builtOutputs = this.getExistingTaskOutputs(taskOutputs);
+            this.removeOutputHashesFromNxOutputs(builtOutputs);
```

Comment:
This has to be moved before we start the command. If the command fails now, the value in the file won't reflect what is in the file system.